Druid/DRStoreTemporaryStagedLoop.class.st

```diff
@@ -0,0 +1,41 @@
+Class {
+	#name : #DRStoreTemporaryStagedLoop,
+	#superclass : #DROptimisation,
+	#category : #'Druid-BytecodeToJITCompilation'
+}
+
+{ #category : #accessing }
+DRStoreTemporaryStagedLoop >> applyTo: cfg [
+
+	cfg phiFunctions
+		select: [ :phi | phi hasRecursiveUse and: [ phi isJITCompileTimeExpression ] ]
+		thenDo: [ :phi | self storeOnNewTemporaryVariables: phi ]
+]
+
+{ #category : #store }
+DRStoreTemporaryStagedLoop >> storeOnNewTemporaryVariables: phiFunction [
```

Comment:
I don't understand what this does :)

Reply:
![image](https://github.com/Alamvic/druid/assets/4098184/801e6600-a0da-4441-bdba-dda6f220fc04)

Hahaha I will add a comment, but I want to improve this in the register allocator. 

Now it is "creating" the variables that iterate the loop, which are variables in the cogit method.
I tried to do it directly in the code generation but doing this is too much simple.